src/lib/drivers/smbus/SMBus.cpp

```diff
@@ -123,7 +124,8 @@ int SMBus::block_read(const uint8_t cmd_code, void *data, const uint8_t length,
 	rx_data[2] = (device_address << 1) | 0x01;
 	byte_count = rx_data[3];
```

Comment:
You should also ensure `byte_count` is not larger than 32, otherwise `get_pec` will read invalid data if we get a corrupt response.
```suggestion
	byte_count = math::min(rx_data[3], 32);
```

Reply:
@bkueng , I think its unnecessary. from what I know, the length of the packet should be constant and equal to 'length'.
and 'length' is limited to be less then 32 anyway.
how 'length' is not equal to 'byte_count', I don't know, its might be due wrong call to the block_read(), that what I asked @flochir 

Note that the block_read() and block_write() are not fully protected as functions, they assume that the caller functions give good arguments.

Reply:
`rx_data[3]` is coming from the transfer, and thus could be anything.

Reply:
Right, `byte_count` (=rx_data[3]) can be anything, but `length` not. this is input to the function that should be well defined. `data[]` length should be in the size of `length`, so limiting the copy to length should be enough, I think.

Reply:
Please read the code: `byte_count` is also used below in `get_pec(rx_data, byte_count + 4);`.

Reply:
@bkueng , got what you meant, anyway it safer.